pkg/policy/selectorcache.go

```diff
@@ -258,6 +264,26 @@ func (s *selectorManager) IsWildcard() bool {
 	return s.key == wildcardSelectorKey
 }
 
+// must be holding selectorcache mutex
+func (s *fqdnSelector) NotifyAdded() {
+	// Make the user (FQDN subsystem) aware of this selector.
+	if s.dnsProxy == nil {
+		return
```

Comment:
The field is not initialized at the time which the `SelectorCache` is initialized (since it is implemented by `RuleGen`, which is bootstrapped later in the DNS-section of Cilium bootstrap). 

To answer your question, given that endpoints which have been restored are not exposed via endpointmanager, and no regenerations specially triggered via the restore process have occurred when we call `SetIdentityPopulator`, this means that there is no way that once we actually start using the `SelectorCache`, that this field can ever be nil. However, we will have the unfortunate fact that the dependency has to be injected later during FQDN bootstrap, leading to somewhat 'incomplete' state in the `SelectorCache` :/ 

I will remove the nil checks, since they can cover incorrect usage of the `SelectorCache`.